test/unit/data_soma/test_soma_adaptor_data_load.py

```diff
@@ -0,0 +1,90 @@
+from test import PROJECT_ROOT
+from test.unit.data_soma.dataset_handler import decompress_dataset
+
+class SomaDataLoadAdaptorTest(unittest.TestCase):
+    """
+    Test file loading, including deferred loading/update.
+    """
+
+    def setUp(self):
+        path = decompress_dataset(f"{PROJECT_ROOT}/example-dataset/tiledb-data/pbmc3k_processed.zip", "pbmc3k_processed")
+        self.data_file = DataLocator(path)
+        config = AppConfig()
+        config.update_server_config(single_dataset__datapath=self.data_file.path)
+        config.update_server_config(app__flask_secret_key="secret")
+        config.complete_config()
+        self.data = SomaAdaptor(self.data_file, config)
+
+    def test_delayed_load_data(self):
+        self.data._create_schema()
+        self.assertEqual(self.data.cell_count, 2638)
+        self.assertEqual(self.data.gene_count, 1838)
+        epsilon = 0.000_005
+        self.assertTrue(self.data.data.X.data.df().to_numpy()[0, 0] - -0.171_469_51 < epsilon)
+
+    def test_diffexp_topN(self):
+        f1 = {"filter": {"obs": {"index": [[0, 500]]}}}
+        f2 = {"filter": {"obs": {"index": [[500, 1000]]}}}
+        result = json.loads(self.data.diffexp_topN(f1["filter"], f2["filter"]))
+        self.assertEqual(len(result["positive"]), 10)
+        self.assertEqual(len(result["negative"]), 10)
+        result = json.loads(self.data.diffexp_topN(f1["filter"], f2["filter"], 20))
+        self.assertEqual(len(result["positive"]), 20)
+        self.assertEqual(len(result["negative"]), 20)
+
+
+class SomaDataLocatorAdaptorTest(unittest.TestCase):
+    """
+    Test various types of data locators we expect to consume
+    """
+
+    def get_basic_config(self):
+        config = AppConfig()
+        config.update_server_config(
+            single_dataset__obs_names=None,
+            single_dataset__var_names=None,
+        )
+        config.update_server_config(app__flask_secret_key="secret")
+        config.update_dataset_config(
+            embeddings__names=["umap"],
+            presentation__max_categories=100,
+            diffexp__lfc_cutoff=0.01,
+        )
+        return config
+
+    def stdAsserts(self, data):
+        """run these each time we load the data"""
+        self.assertIsNotNone(data)
+        self.assertEqual(data.cell_count, 2638)
+        self.assertEqual(data.gene_count, 1838)
+
+    def test_posix_file(self):
+        path = decompress_dataset(f"{PROJECT_ROOT}/example-dataset/tiledb-data/pbmc3k_processed.zip", "pbmc3k_processed")
+        locator = DataLocator(path)
+        self.assertEqual(SomaAdaptor.file_size(locator), 480)
+        config = self.get_basic_config()
+        config.update_server_config(single_dataset__datapath=locator.path)
+        config.complete_config()
+        data = SomaAdaptor(locator, config)
+        self.stdAsserts(data)
+
+    # TODO: using SOMA with a TileDB URI is not tested yet
+    # def test_url_https(self):
```

Comment:
You can remove these tests

Reply:
I've removed it for now, but won't we ever be loading SOMA data from a URL like we can with .h5ad?